lib/tune.cpp

```diff
@@ -718,82 +718,80 @@ namespace quda {
 	 can't guarantee that all nodes are partaking */
       if (comm_rank() == 0 || !commGlobalReduction() || policyTuning()) {
 	TuneParam best_param;
-	cudaError_t error = cudaSuccess;
-	cudaEvent_t start, end;
-	float elapsed_time, best_time;
-	time_t now;
+        qudaError_t error = qudaSuccess;
```

Comment:
Having thought about this a bit, I do wonder if it would make more sense to make the tune.cpp file a CUDA specific thing, and have different implementations for other architectures.  Much of the tuning aspects in here are CUDA specific, and do not make sense to other platforms.  E.g., 3-d block and grid tuning doesn't really apply to OpenMP or pSTL.

Thoughts?  Perhaps simply rename this file to tune_cuda.cpp and add different implementations in new files for other targets?  There has also been the thought that the QUDA autotuner is spun out as its own separate thing, and bake in QUDA abstractions would fly in the face of that.

Reply:
Agree. The tuning as it stands in very a much thread/block oriented. OpenMP or (ACC) may wish to use different tiling in collapsed loops.